tests/tests_pytorch/tuner/test_lr_finder.py

```diff
@@ -415,3 +416,15 @@ def __init__(self):
     lr_finder = trainer.tuner.lr_find(model=model, update_attr=True, num_training=1)  # force insufficient data points
     assert lr_finder.suggestion() is None
     assert model.learning_rate == 0.123  # must remain unchanged because suggestion is not possible
+
+
+def test_lr_finder_with_stochastic_weight_averaging(tmpdir):
+    """Regression test for issue https://github.com/Lightning-AI/lightning/issues/14755"""
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.learning_rate = 0.123
+
+    model = TestModel()
+    trainer = Trainer(default_root_dir=tmpdir, callbacks=[StochasticWeightAveraging(swa_lrs=0.01)], auto_lr_find=True)
+    trainer.tune(model)
```

Comment:
I have no idea what to assert here lol.

Reply:
One that does:

```python
swa = StochasticWeightAveraging()
assert swa._average_model is None
```

would be enough. This is only uncovered during tuning because we save a checkpoint before `setup` has been called. In `setup` we write this attribute again which explains why tests didn't catch this before.

The bug is so simple that we could not test it too. It's unfortunate that it was missed in code review. This bug reminds me of the one in https://github.com/Lightning-AI/lightning/pull/14809. We need to be stricter when reviewing typing changes.

Reply:
Added better test

Reply:
Thanks for the suggestion